src/protocol/mod.rs

```diff
@@ -102,6 +102,23 @@ impl Step {
             id: self.id.clone() + "/" + step.as_ref(),
         }
     }
+
+    /// Determine if this step is "narrower" than the `other` step.
```

Comment:
I don't think that this is going to be needed, assuming that the above design works.